src/bin/rav1e.rs

```diff
@@ -29,31 +29,42 @@ use crate::decoder::VideoDetails;
 use std::fs::File;
 use std::io::BufWriter;
 
-fn read_frame<T: Pixel, D: Decoder>(ctx: &mut Context<T>, decoder: &mut D, video_info: VideoDetails) {
-  match decoder.read_frame(&video_info) {
-    Ok(frame) => {
-      match video_info.bit_depth {
-        8 | 10 | 12 => {}
-        _ => panic!("unknown input bit depth!")
-      }
+struct Source<D: Decoder> {
+ limit: usize,
+ input: D,
+}
 
-      let _ = ctx.send_frame(Some(Arc::new(frame)));
-    }
-    _ => {
+impl<D: Decoder> Source<D> {
+  fn read_frame<T: Pixel>(&mut self, ctx: &mut Context<T>, video_info: VideoDetails) {
+    if self.limit == 0 {
```

Comment:
This seems not to fit well with the default value of zero, which indicates to read until the end of the input stream. Unless I'm mistaken, in this case no frames are read.

Reply:
Good catch!